### PR TITLE
Addition of `FloatScalingFilter`

### PR DESCRIPTION
**Pull request by nguyenv**
- Fix types for setting and getting options
- Do not actually set option defaults to -1

### CODE REVIEW

**Comment by ihnorton** [comment]
@abigalekim we need to sync up on this and figure out what the issue is here.

File: tiledb/tests/test_filters.py [hunk 1]
```diff
@@ -60,6 +62,7 @@ def test_all_filters(self):
             tiledb.PositiveDeltaFilter(),
             tiledb.ChecksumSHA256Filter(),
             tiledb.ChecksumMD5Filter(),
+            tiledb.FloatScalingFilter(1, 0),
```
[the remark this answers is not in the document]
Comment (nguyenv) [reply tiledb/tests/test_filters.py:65]: I had bitwidth defaulted to 8 but can modify that to match core.

File: tiledb/filter.py [hunk 2]
```diff
@@ -477,6 +527,117 @@ def _attrs_(self):
...
+        self._bytewidth = bytewidth
+        self._ctx = ctx or default_ctx()
+
+        super().__init__(lt.FilterType.SCALE_FLOAT, self._ctx)
+
+        if factor:
+            self._set_option(
+                lt.Context(self._ctx.__capsule__(), False),
+                lt.FilterOption.SCALE_FLOAT_FACTOR,
+                float(factor),
```
Comment (nguyenv) [inline tiledb/filter.py:595]: (BTW I will be cleaning up the code and adding better type checking for all this eventually.)

**Review by ihnorton — COMMENTED** [review]
Noting for merge check: if an unsupported value for bytewidth is put through the int32 signature calling `tiledb_filter_set_option`, then the exception thrown inside the FloatScaleFilter constructor is currently suppressed. (not sure whether that's happening in -py or core, but this led to a lot of confusion)

File: tiledb/__init__.py [hunk 3]
```diff
@@ -76,6 +76,7 @@
     PositiveDeltaFilter,
     ChecksumMD5Filter,
     ChecksumSHA256Filter,
+    FloatScalingFilter,
```
[the remark this answers is not in the document]
Comment (ihnorton) [reply tiledb/__init__.py:79]: but `Scaling` does sound better so let's stick with that.

File: tiledb/filter.py [hunk 4]
```diff
@@ -477,6 +517,124 @@ def _attrs_(self):
...
+    """
+
+    def __init__(
+        self,
+        factor: float = None,
+        offset: float = None,
+        bytewidth: int = None,
+        ctx: "Ctx" = None,
+    ):
+        # try:
```
Comment (ihnorton) [inline tiledb/filter.py:561]: Remove?

**Review by ihnorton — APPROVED** [review]
LGTM aside from some minor nits. Thanks!

**Pull request merged**